### PR TITLE
Add option to sort translations by msgid

### PR DESCRIPTION
**Pull request by angelikatyborska**
I would like to propose this feature to elixir-gettext/gettext. Is there something I can improve so that it's more likely to be accepted?

We want to try having the translations sorted because always appending translations to the end makes it certain that there will be merge conflicts when translations are added on two separate branches at the same time. This happens to us literally every day.

For our own purposes, I created the branch  feature/sort-msgids-v0.17.1 which has commits from this branch cherry-picked onto the tag v0.17.1. This is because master already contains breaking changes that break our main app and Timex.

### CODE REVIEW

File: lib/gettext/extractor.ex [hunk 1]
```diff
@@ -298,8 +298,22 @@ defmodule Gettext.Extractor do
     # with the translations that only appear in `new`.
     unique_new = Enum.reject(new.translations, &PO.Translations.find(existing.translations, &1))
 
+    translations = old_and_merged ++ unique_new
+
+    translations =
+      if gettext_config[:sort_by_msgid] do
+        Enum.sort_by(translations, fn t ->
```
Comment (mtarnovan) [inline lib/gettext/extractor.ex]: @angelikatyborska `Enum.sort_by(translations, &(&1.msgid))`?

**Comment by angelikatyborska** [comment]
I thought about allowing sorting by more than just msgid, but I couldn't really decide what else it could be sorted by? Comments, references? Definitely not msgstr because that differs by language.

**Comment by dirkholzapfel** [comment]
> I thought about allowing sorting by more than just msgid, but I couldn't really decide what else it could be sorted by? Comments, references? Definitely not msgstr because that differs by language.

I think `msgid` is the only stable key in this file

**Comment by angelikatyborska** [comment]
Thanks for the reviews. I'm closing this PR because we don't really need it, it was just to get a review. I set myself a reminder in 2 weeks to open a PR against the main fork of gettext.

**Pull request closed without merging**